### PR TITLE
Add --force to podman umount to force the unmounting of the rootfs

### PR DESCRIPTION
**Pull request by rhatdan**
podman umount will currently only unmount file system if not other
process is using it, otherwise the umount decrements the container
storage to indicate that the caller is no longer using the mount
point, once the count gets to 0, the file system is actually unmounted.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes https://github.com/projectatomic/libpod/issues/1183

File: libpod/container_internal.go [hunk 1]
```diff
@@ -1332,9 +1332,9 @@ func (c *Container) mount() (string, error) {
 }
 
 // unmount unmounts the container's root filesystem
-func (c *Container) unmount() error {
+func (c *Container) unmount(force bool) error {
 	// Also unmount storage
```
Comment (mheon) [inline libpod/container_internal.go]: Force should not be allowed if the container if in ContainerStateRunning or ContainerStatePaused, or if there are one or more active exec sessions - we don't want people force-unmounting a running container.

File: cmd/podman/umount.go [hunk 2]
```diff
@@ -16,13 +16,23 @@ var (
...
 			Usage: "umount all of the currently mounted containers",
 		},
+		cli.BoolFlag{
+			Name:  "force, f",
+			Usage: "force the complete umount all of the currently mounted containers",
+		},
 	}
 
+	description = `
+Container storage keeps track of the number of times the container was mounted.
```
Comment (mheon) [inline cmd/podman/umount.go]: Suggest rewording: "Decrement the mount counter for a container, unmounting the container if the counter reaches 0. Unmounting can be force with the --force flag."

File: docs/podman-umount.1.md [hunk 3]
```diff
@@ -7,13 +7,24 @@ podman\-umount - Unmount the specified working containers' root file system.
...
 ## OPTIONS
 **--all, -a**
 
 All of the currently mounted containers will be unmounted.
 
+**--force, -f**
+
+Force the unmouning of specified containers' root file system, even if other
+processes have mounted it.
+
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/podman-umount.1.md]: I can add it, but the same issue can happen if you repeatedly remove the image.

**Comment by rhatdan** [comment]
BTW I also added a error if you try to umount an image that was not mounted.  Previously you could keep umounting and it would just print the CID, even if the container wasn't mounted.

File: docs/podman-umount.1.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply docs/podman-umount.1.md]: I'm always on the fence for these kind of things.  Let's leave it as you had it unless others have strong leanings.  We can always add a warning later if we find people getting tripped up by it.

[the remark this answers is not in the document]
Comment (edsantiago) [reply docs/podman-umount.1.md]: Typo: "umouning"

File: libpod/container_internal.go (hunk 1 above)
Comment (mheon) [reply libpod/container_internal.go]: Good catch - nice to know we were thinking of these kind of problems when we wrote it originally.

File: docs/podman-umount.1.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/podman-umount.1.md]: Fixed

File: cmd/podman/umount.go [hunk 4]
```diff
@@ -16,13 +16,23 @@ var (
 			Name:  "all, a",
 			Usage: "umount all of the currently mounted containers",
 		},
+		cli.BoolFlag{
+			Name:  "force, f",
+			Usage: "force the complete umount all of the currently mounted containers",
```
Comment (edsantiago) [inline cmd/podman/umount.go]: I'm not sure I grok this. It looks like you copypasted from --all but didn't finish writing the description?

**Comment by mheon** [comment]
LGTM aside from the comment from @edsantiago

File: cmd/podman/umount.go [hunk 5]
```diff
@@ -36,6 +46,7 @@ func umountCmd(c *cli.Context) error {
 	}
 	defer runtime.Shutdown(false)
 
+	force := c.Bool("force")
```
Comment (edsantiago) [inline cmd/podman/umount.go]: This is probably a stupid question. In some places, when a 1-letter shortcut is defined for an option, I see:
```
    x := c.Bool("longname") || c.Bool("x")
```
(Seems to be more prevalent with `c.IsSet` than `Bool`). I just want to make sure that the above line also recognizes `-f`. No need to respond if it does.

**Comment by edsantiago** [comment]
LGTM other than the usage message. I like the new check for mountedness.

File: cmd/podman/umount.go [hunk 6]
```diff
@@ -16,13 +16,23 @@ var (
...
+			Usage: "force the complete umount all of the currently mounted containers",
+		},
 	}
 
+	description = `
+Container storage keeps track of the number of times the container was mounted.
+Unmounting a container will decrement the counter, and actually unmount it when
+no other processes are using it.  You can force the unmount with the --force
+command.
+`
```
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply cmd/podman/umount.go]: I'm not seeing the "Here is what I have now" wording.  Did you decide to stick with the original or do you have a merge hiccup?

**Comment by rhatdan** [comment]
@TomSweeneyRedHat You were right, I also fixed another issue, I had where we were reporting lots of errors when you did a 
podman umount --all
Now I ignore not mounted errors when you specify --all

@edsantiago the CLI works with either --force of -f, the Bool("force") gets set for both.  I think these other cases are actually "bugs"

**Comment by rhatdan** [comment]
@rh-atomic-bot r=TomSweeneyRedHat
